hs/src/Reach/Eval/Core.hs

```diff
@@ -891,7 +891,7 @@ applyType ct v = \case
 typeCheck_s :: ClaimType -> SLType -> SLVal -> App DLArgExpr
 typeCheck_s ct st val = do
   dt <- st2dte st
-  res <- typeCheck_d dt val
+  res <- typeCheck_d Nothing dt val
```

Comment:
You could make this take an argument. I think you should define `type MSrcLoc = Maybe SrcLoc`, because it is so brutal to type out